client/src/panels/panel_declarations/results/IndicatorPanel.js

```diff
@@ -102,56 +101,46 @@ const query_api = (id) => {
 export default class IndicatorPanel extends React.Component {
   constructor(props) {
     super(props);
-    this.state = { loading: true };
+    this.state = { loading: true, show_modal: false };
   }
 
   componentDidMount() {
-    const {
-      match: {
-        params: { id },
-      },
-    } = this.props;
+    const { id } = this.props;
     query_api(id).then(() => this.setState({ loading: false }));
   }
```

Comment:
This is going to fire for every "`IndicatorPanel`" as soon as the button mounts in the table. That intentional?

In the best case, that means the loading happens in the background and the user never sees a spinner... but it might also mean that the one indicator they cared about is stuck in a long loading queue behind a bunch of other API queries that they don't actually care about and they get stuck with a spinner longer than they wanted. In the worst (and probably most likely case), we're burning up their network/device resources on a ton of network traffic and JS processing for resources they'll never see.

I'd say don't kick off the loading till the given modal is opened.